src/utilities/filetypes/EpwFile.cpp

```diff
@@ -567,79 +568,271 @@ boost::optional<EpwDesignCondition> EpwDesignCondition::fromDesignConditionsStri
+  if (itCooling == strings.cend()) {
+    LOG_FREE(Error, "openstudio.EpwFile", "Could not find 'Cooling' design condition in EPW design condition string");
+    return boost::none;
+  }
+  auto itExtreme = std::find_if(strings.cbegin(), strings.cend(), [](const auto& s) { return openstudio::istringEqual(s, "Extremes"); });
+  if (itExtreme == strings.cend()) {
+    LOG_FREE(Error, "openstudio.EpwFile", "Could not find 'Extremes' design condition in EPW design condition string");
+    return boost::none;
+  }
+
+  std::span<const std::string> heatingStrings(std::next(itHeating, 1), itCooling);
+  std::span<const std::string> coolingStrings(std::next(itCooling, 1), itExtreme);
+  std::span<const std::string> extremesStrings(std::next(itExtreme, 1), strings.cend());
+
+  const auto numHeatingFields = heatingStrings.size();
+  const auto numCoolingFields = coolingStrings.size();
+  const auto numExtremeFields = extremesStrings.size();
+
+  // Default to 2009 for now
+  // constexpr size_t numOtherFields = 5;  // Title, Blank, Heating, Cooling, Extreme
+  size_t expectedNumHeatingFields = 15;
+  const size_t expectedNumCoolingFields = 32;
+  size_t expectedNumExtremeFields = 16;
+
+  bool needHeuristics = false;
+
+  if (assumedVersion) {
+    if (*assumedVersion <= ASHRAEHoFVintage::ASHRAE_2017) {
+    } else {
+      expectedNumHeatingFields = 16;
+    }
+    if (*assumedVersion <= ASHRAEHoFVintage::ASHRAE_2013) {
+      expectedNumHeatingFields = 15;
+      expectedNumExtremeFields = 16;
+    } else if (*assumedVersion <= ASHRAEHoFVintage::ASHRAE_2017) {
+      expectedNumExtremeFields = 15;
+      expectedNumExtremeFields = 15;
+    } else {
+      expectedNumExtremeFields = 16;
+      expectedNumExtremeFields = 15;
+    }
+    // Check that the field counts match the assumed version, we'll fall back to heuristics if not
+    if (numHeatingFields != expectedNumHeatingFields) {
+      LOG_FREE(Warn, "openstudio.EpwFile",
+               "Based on the assumed vintage parsed in the title string of " << assumedVersion->valueDescription() << ", we expected "
+                                                                             << expectedNumHeatingFields << " Heating fields but got "
+                                                                             << numHeatingFields << ", falling to back to heuristics.");
+      needHeuristics = true;
+    }
+    if (numExtremeFields != expectedNumExtremeFields) {
+      LOG_FREE(Warn, "openstudio.EpwFile",
+               "Based on the assumed vintage parsed in the title string of " << assumedVersion->valueDescription() << ", we expected "
+                                                                             << expectedNumExtremeFields << " Extreme fields but got "
+                                                                             << numExtremeFields << ", falling to back to heuristics.");
+      needHeuristics = true;
+    }
+  } else {
+    needHeuristics = true;
+  }
```

Comment:
If there was a parsable version in the title, we check that the number of fields are matching what we expect. If not, we warn and still fall back to using heuristics.